File: src/confy.nim
#:______________________________________________________________________
#  ᛝ confy  |  Copyright (C) Ivan Mar (sOkam!)  |  GNU GPLv3 or later  :
#:______________________________________________________________________
##! @fileoverview
##!
##!  Confy : Comfortable and Configurable Buildsystem
##!  @todo : Docs here
#_____________________________________________________|
from ./confy/target        import nil ; export target
from ./confy/package       import nil ; export package
from ./confy/dependency    import nil ; export dependency
from ./confy/flags         import nil ; export flags
from ./confy/tools/version import nil ; export version
from ./confy/tools/git     import nil ; export git
from ./confy/tools/files   import nil ; export files
from ./confy/tools/args    import nil ; export args
from ./confy/tools/shell   import nil ; export shell
from ./confy/log           import nil ; export log
from ./confy/command       import nil ; export command
from ./confy/systm as sys  import nil ; export sys
from ./confy/types import nil
from ./confy/types/base import Name
export Name
export types.Config
export types.System
export types.OS ; export types.CPU ; export types.ABI
export types.build.Mode; export types.build.ModeKind; export types.build.ModeOptim
from ./confy/state as G import nil ;
export G.cfg ; export G.cli

#_______________________________________
# @section Forward Export useful std tools
#_____________________________
from std/os import nil
export os.`/`
export os.fileExists
export os.dirExists
export os.getEnv
export os.parentDir
export os.removeDir

File: src/confy.zig
//:______________________________________________________________________
//  ᛝ confy  |  Copyright (C) Ivan Mar (sOkam!)  |  GNU LGPLv3 or later  :
//:______________________________________________________________________
//! @fileoverview
//!  Confy : Comfortable and Configurable Buildsystem
//!  @todo : Docs here
//_____________________________________________________|
const confy = @This();

//______________________________________
// @section Forward Export zstd tools
//____________________________
const zstd = @import("./lib/zstd.zig");
pub const Name      = zstd.Name;
pub const echo      = zstd.echo;
pub const cstr      = zstd.cstr;
pub const cstr_List = zstd.cstr_List;
pub const Lang      = zstd.Lang;
pub const System    = zstd.System;


//______________________________________
// @section Forward Export Confy tools
//____________________________
pub const init      = @import("./confy/core.zig").init;
pub const Git       = @import("./confy/git.zig");
pub const Package   = @import("./confy/package.zig");
pub const BuildTrg  = @import("./confy/target.zig");
pub const Program   = BuildTrg.Program;
pub const StaticLib = BuildTrg.StaticLib;
pub const SharedLib = BuildTrg.SharedLib;
pub const UnitTest  = BuildTrg.UnitTest;
pub const CodeList  = @import("./confy/code.zig");
pub const FlagList  = @import("./confy/flags.zig");

